### PR TITLE
[openal-soft] Update OpenAL Soft to 1.20.0

### PR DESCRIPTION
**Pull request by dhkatz**
Update OpenAL Soft to its latest version.

I tried to replicate the same fixes the patches were trying to do before, as 1.20.0 made some changes to its CMake files. Just need to see what the CI says.

### CODE REVIEW

File: ports/openal-soft/portfile.cmake [hunk 1]
```diff
@@ -6,8 +6,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO kcat/openal-soft
-    REF openal-soft-1.19.1
-    SHA512 4a64cc90ddeaa3773610b0bc8023d231100f3396f3fc5bd079db81600f80a789c75e6af03391bfc78a903c96bb71f8052a9ae802ea81422028e5b12b7eb6c47b
+    REF openal-soft-1.20.0
+    SHA512 d106bf8f96b32a61fadc0ee54882ce5041e4cbc35bf573296a210c83815b6c7be056ee3ed7617196dda5f89f2acd7163375f14b0cf24934faa0eda1fdb4f82a9
     HEAD_REF master
     PATCHES
         dont-export-symbols-in-static-build.patch
```
Comment (NancyLi1013) [inline ports/openal-soft/portfile.cmake]:
```
if(VCPKG_CMAKE_SYSTEM_NAME)
    set(ALSOFT_REQUIRE_WINDOWS OFF)
    set(ALSOFT_REQUIRE_LINUX ON)
else()
    set(ALSOFT_REQUIRE_WINDOWS ON)
    set(ALSOFT_REQUIRE_LINUX OFF)
endif()
```
Maybewe can use `VCPKG_TARGET_IS_WINDOWS` and `VCPKG_TARGET_IS_LINUX`  instead.

Comment (NancyLi1013) [reply ports/openal-soft/portfile.cmake]: Could you update this
```
file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/openal-soft)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/openal-soft/COPYING ${CURRENT_PACKAGES_DIR}/share/openal-soft/copyright)
```
as `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` ?

Comment (NancyLi1013) [reply ports/openal-soft/portfile.cmake]: `pkgconfig` is needed on Linux platform.
So we should not remove it from `lib` or `debug/lib`  directory.
```
file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/pkgconfig)
file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/openal-soft/portfile.cmake (hunk 1 above)
Comment (dhkatz) [reply ports/openal-soft/portfile.cmake]: These were all changes made be previous port maintainers but I'll go ahead and fix them per your recommendations.

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**